pkg/agents/baseagent.go

```diff
@@ -0,0 +1,66 @@
+package agents
+
+import (
+	"errors"
+	"log"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra/messages"
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra/tower"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/abm"
+)
+
+type Base struct {
+	id    string
+	tower *tower.Tower
+}
+
+func NewBaseAgent(abm *abm.ABM, uuid string) (*Base, error) {
+	world := abm.World()
+	if world == nil {
+		return nil, errors.New("Agent needs a World defined to operate")
+	}
+	tower, ok := world.(*tower.Tower)
+	if !ok {
+		return nil, errors.New("Agent needs a Tower world to operate")
+	}
+	return &Base{
+		tower: tower,
+		id:    uuid,
+	}, nil
+}
+
+func (a *Base) Run() {
+	floor := a.tower.Floor(a.id)
+	log.Printf("An agent cycle executed from base agent %d", floor)
+}
+
+func (a *Base) HP() int {
+	return a.tower.HP(a.id)
+}
+
+func (a *Base) Floor() int {
+	return a.tower.Floor(a.id)
+}
+
+func (a *Base) ID() string {
```

Comment:
very minor, but maybe should name the HP() function 'get_HP()' just to make it a little clearer. Same for Floor() and ID().

Reply:
Please see this for why it is just `HP()`: https://go.dev/doc/effective_go#Getters